### ISSUE TITLE
argocd admin dasboard --address isn't used for web c

### ISSUE THREAD

**Issue by idletea** [argoproj/argo-cd#17152]
This is essentially re-opening #16379 as I'm still experiencing the issue on a recent build.

`argocd admin dashboard` allows the host and port bound to to be set by flags, but the host isn't propagated to all relevant systems, rather `127.0.0.1` appears to be hardcoded.

At a glance: the issue is specifically for requests to the `/api` endpoints, for example `GET http://127.8.0.10/api/version` happens on page load, but returns a 503 with the error in the screenshot below.

**To Reproduce**

Run the argocd dashboard with loopback address other than `127.0.0.1` like so:
```
$  argocd admin dashboard --address 127.8.0.10 --port 80
```

and visit the main page at `http://127.8.0.10/`

**Expected behavior**

All resources connect using the address given.

**Screenshots**

![20240208_15h53m54s_grim](https://github.com/argoproj/argo-cd/assets/8740147/8f920e9c-cbf7-4c7f-b391-33069205f95f)

**Version**

```shell
argocd: v2.10.0+2175939
  BuildDate: 2024-02-06T14:47:59Z
  GitCommit: 2175939ed6156ddd743e60f427f7f48118c971bf
  GitTreeState: clean
  GoVersion: go1.21.6
  Compiler: gc
  Platform: linux/amd64
```

**Comment by todaywasawesome** [comment]
Hmm, I see the commit [2f2958a](https://github.com/argoproj/argo-cd/commit/2f2958a0f42068552de67370caccb84522267c65) is in all the 2.10 releases (including RCs) so I would think it would be fixed.

**Comment by todaywasawesome** [comment]
https://github.com/argoproj/argo-cd/blob/7e80f1e8e59da2701c87e97b6bda4c64345a90e2/pkg/apiclient/apiclient.go#L208-L224

Are you using any port forwarding? The only time it should get a hardcoded 127.0.0.1 address is if port-forwarding is happening.

**Comment by idletea** [comment]
@todaywasawesome Nope, no forwarding. I put the `CAP_NET_BIND_SERVICE`  capability on argocd so it can bind to 80 directly.

Incidentally the port used does get passed along - if I do `--port 8000` instead the error indicates `127.0.0.1:8000` is what's not responding.

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and let us know if the issue is still present, please?
